rpcs3/Emu/RSX/rsx_cache.h

```diff
@@ -643,17 +656,22 @@ namespace rsx
 			}
 
 			pipeline_data data = pack(pipeline, vp, fp);
+
 			std::string fp_name = root_path + "/raw/" + fmt::format("%llX.fp", data.fragment_program_hash);
 			std::string vp_name = root_path + "/raw/" + fmt::format("%llX.vp", data.vertex_program_hash);
 
 			if (!fs::is_file(fp_name))
 			{
-				fs::file(fp_name, fs::rewrite).write(fp.get_data(), fp.ucode_length);
+				const std::string temp_name = fmt::format("%s.%s.temp", fp_name, fmt::base57(utils::get_unique_tsc()));
```

Comment:
This pattern appears a lot. Refactor it to some common utility in file or the cache class and avoid repetition.

Reply:
Fixed.